src/inference/src/dev/threading/istreams_executor.cpp

```diff
@@ -553,109 +554,113 @@ IStreamsExecutor::Config IStreamsExecutor::Config::reserve_cpu_threads(const ISt
     return config;
 }
 
-void IStreamsExecutor::Config::update_executor_config(int stream_nums,
-                                                      int threads_per_stream,
-                                                      IStreamsExecutor::Config::PreferredCoreType core_type,
-                                                      bool cpu_pinning) {
-    const auto proc_type_table = ov::get_proc_type_table();
+void IStreamsExecutor::Config::update_executor_config() {
+    const auto proc_type_table = get_proc_type_table();
+    bool streams_info_available = false;
 
     if (proc_type_table.empty()) {
         return;
     }
 
-    if (proc_type_table.size() > 1) {
-        core_type = ov::threading::IStreamsExecutor::Config::ANY;
+    if (!_streams_info_table.empty()) {
+        streams_info_available = true;
+        std::vector<int> threads_proc_type(HYPER_THREADING_PROC + 1, 0);
+        for (size_t i = 0; i < _streams_info_table.size(); i++) {
+            if (_streams_info_table[i][NUMBER_OF_STREAMS] > 0) {
+                threads_proc_type[_streams_info_table[i][PROC_TYPE]] +=
+                    _streams_info_table[i][THREADS_PER_STREAM] * _streams_info_table[i][NUMBER_OF_STREAMS];
+            }
+        }
+        for (size_t i = ALL_PROC; i < threads_proc_type.size(); i++) {
+            if (threads_proc_type[i] > proc_type_table[0][i]) {
+                streams_info_available = false;
+                break;
+            }
+        }
     }
 
-    // IStreamsExecutor::Config config = initial;
-    const auto total_num_cores = proc_type_table[0][ALL_PROC];
-    const auto total_num_big_cores = proc_type_table[0][MAIN_CORE_PROC] + proc_type_table[0][HYPER_THREADING_PROC];
-    const auto total_num_little_cores = proc_type_table[0][EFFICIENT_CORE_PROC];
+    if (_streams_info_table.empty() || !streams_info_available) {
+        _streams_info_table.clear();
```

Comment:
Do we need `clear()` if `_streams_info_table.empty()`

Reply:
Updated. `clear` if `!streams_info_available`.